readme.md

```diff
@@ -412,6 +412,19 @@ const createBearSlice = (set, get) => ({
 })
 ```
 
+You can also log action's type along with its payload:
+
+```jsx
+const createBearSlice = (set, get) => ({
+  addFishes: (count) =>
+    set(
+      (prev) => ({ fishes: prev.fishes + count }),
+      false,
+      { type: 'bear/addFishes', payload: count }
```

Comment:
I think it's good to have the payload key something other than "payload" to imply "payload" is not a special key, it can be anything...

```suggestion
      { type: 'bear/addFishes', count }
```

Reply:
It's Redux-style logging and the payload is a special key for action in Redux Devtools. IMO it would be great to keep it like that while using Redux Devtools.
![image](https://user-images.githubusercontent.com/9741188/184088957-c34300f4-c501-4008-8860-5309868ea2fb.png)
Here is a link to the conversation on the pmndrs DC: https://discord.com/channels/740090768164651008/740093228904218657/1006824679111135343

Reply:
In what way is it special? Like how does redux devtools treat `payload` key different from `count` key.

Reply:
The main goal of this PR is logging the action's payload as the `payload` field, not any field name the user wants. To be consistent with the Redux convention (https://github.com/redux-utilities/flux-standard-action).
Ok, I am getting now that you want to abstract from it, right? The question is whether we want to have it consistent or to have whatever field's name the user wants but I think we should leave this decision for Zustand's owner.

Reply:
Okay so as I guessed redux devtools doesn't treat payload key in any special way.

Users allowed to choose the convention they want to follow, the one you've linked is just one of the many styles out there, it's not required. The only requirement and constraint of a redux action is that [it must have a `type` property](https://github.com/reduxjs/redux/blob/bb6233b726b2fda2b503364135899274ccc8973c/src/types/actions.ts#L1-L31).

I understand that you want to impose an convention for your logger, but the code being edited here affects everyone, and we can't impose a non-required convention for everyone.

> I think we should leave this decision for Zustand's owner

Sure, just giving my two cents.

Reply:
I don't want to impose some of my conventions. I assumed that is a well-known standard and it would be nice to be consistent with it, but not. And it's okay.

I will add `[key: string]: unknown;` but you mention:
> in general it's good to stay away from index signatures

Could you elaborate? Is there the other way to achieve that?

Reply:
Ah I missed this comment, it's mostly resolved now but I'll reply for sake of completeness...

> I don't want to impose some of my conventions. I assumed that is a well-known standard and it would be nice to be consistent with it, but not. And it's okay.

I never said it was "your" convention, I'm well aware about FSA, and not everyone uses it. My point was simply this: Redux does not require users to follow FSA so we should also not.

> Could you elaborate? Is there the other way to achieve that?

Now I did here https://github.com/pmndrs/zustand/pull/1183#issuecomment-1213132016